api/v2/checluster_types.go

```diff
@@ -541,6 +546,54 @@ type PodSecurityContext struct {
 	FsGroup *int64 `json:"fsGroup,omitempty"`
 }
 
+type CheClusterGitServices struct {
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="GitHub"
+	GitHub []GitHubService `json:"github,omitempty"`
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="GitLab"
+	GitLab []GitLabService `json:"gitlab,omitempty"`
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="BitBucket"
+	BitBucket []BitBucketService `json:"bitbucket,omitempty"`
+}
+
+// GitHubService enables users to work with a remote Git repository that is hosted on GitHub.
+type GitHubService struct {
+	// Kubernetes secret, that contains Base64-encoded GitHub OAuth Client id and GitHub OAuth Client secret data.
+	// The GitHub OAuth data must be stored in the Kubernetes secret in `id` and `secret` keys respectively.
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:Secret"
+	SecretName string `json:"secretName"`
+}
+
+type GitLabService struct {
+	// Kubernetes secret, that contains Base64-encoded GitHub Application id and GitHub Application Client secret data.
+	// The GitHub Application data must be stored in `id` and `secret` keys of the Kubernetes secret respectively.
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:Secret"
+	SecretName string `json:"secretName"`
+	// GitLab server endpoint URL.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default:="https://gitlab.com"
+	Endpoint string `json:"endpoint"`
+}
+
+type BitBucketService struct {
+	// Kubernetes secret, that contains Base64-encoded BitBucket OAuth 1.0 or OAuth 2.0 data.
+	// For OAuth 1.0: private key, BitBucket Application link consumer key and BitBucket Application link shared secret must be stored
+	// in `private.key`, `consumer.key` and `shared_secret` keys of the Kubernetes secret respectively.
+	// For OAuth 2.0: BitBucket OAuth consumer key and BitBucket OAuth consumer secret must be stored
+	// in `id` and `secret` keys of the Kubernetes secret respectively.
```

Comment:
```suggestion
	// in `id` and `secret` keys of the Kubernetes secret respectively.
	// See [Eclipse Che Administration Guide](https://www.eclipse.org/che/docs/stable/administration-guide/configuring-oauth-1-for-bitbucket/) for details.
```
@tolusha @vinokurig I don't see OAuth 2 bitbucket in Che documentation.

Reply:
There is a PR for it https://github.com/eclipse-che/che-docs/pull/2410